ldi-core/ldes-client/tree-node-fetcher/src/main/java/ldes/client/treenodefetcher/domain/valueobjects/TreeNodeRequest.java

```diff
@@ -25,7 +25,7 @@ public Request createRequest() {
 		RequestHeaders requestHeaders = new RequestHeaders(
 				List.of(new RequestHeader(HttpHeaders.ACCEPT, lang.getHeaderString())));
 		if (etag != null) {
-			requestHeaders = requestHeaders.addRequestHeader(new RequestHeader(HttpHeaders.IF_NONE_MATCH, etag));
+			requestHeaders = requestHeaders.withRequestHeader(new RequestHeader(HttpHeaders.IF_NONE_MATCH, etag));
```

Comment:
any difference?

Reply:
Yes, with 'add' I didn't expect to get a new list so I wrote code calling this method assuming the return value was void or a boolean (similar to the add method on java lists). Then I realized my code was not working because the implementation returns a new list with the added header. Using 'with' better portrays this behavior